interview-questions.md

```diff
@@ -8,52 +8,60 @@ Then, research the question to expand on your answer. Even if you feel you have
 1. What is a hash in Ruby?
 
   Your answer:
+  A hash in ruby is a collection of key:value pairs. 
 
   Researched answer:
-
+  A hash in ruby is a collection of key:value pairs. A hash is similar to an array except taht indexing is done via arbitrary keys of an object type and not trough integer index.If you attempt to access a hash with a key that does not exist, the method will return nil.
 
 
 2. What is a gem?
 
   Your answer:
+  Gems are used to modify and even speed up code in ruby applications.
 
   Researched answer:
-
-
+  Gems are equivalent to a plugin or an extension for the Ruby language programming. Gems can be used to extend or modify functionality in Ruby applications. Gems can be created and sharded amongst developers through a shared community to be used in their applications and libraries. Some gems provide command line utilities to help automate tasks and speed up your work.
 
 3. What is Ruby on Rails?
 
   Your answer:
+  Ruby on Rails is a web application framework written in Ruby. Rails is a MVC framework.
 
   Researched answer:
-
-
+  Rails is a model–view–controller (MVC) framework, providing default structures for a database, a web service, and web pages. Rails facilitates the use of web standards such as JSON or XML for data transfer and HTML, CSS and JavaScript for user interfacing. Rails can be used to build web applications taht span both the front end and back end. The Model and Controller would be considered the back-end part of the application. The View handles the front end by generating the actual web page that people see in the browser, along with included JavaScript and HTML. You can also use Rails to create web services or APIs that return JSON for use by other applications.
```

Comment:
Some lines look very similar to the following article: https://www.hashlabs.com/services/ruby-on-rails/#:~:text=Rails%20is%20a%20model%E2%80%93view,for%20display%20and%20user%20interfacing